src/main/java/uk/ac/ebi/eva/commons/core/models/VariantSource.java

```diff
@@ -76,7 +76,9 @@ public VariantSource(String fileId, String fileName, String studyId, String stud
         this.date = date;
         this.samplesPosition = samplesPosition;
         this.metadata = metadata;
-        this.stats = stats;
+        if (stats != null) {
+            this.stats = new VariantGlobalStats(stats);
```

Comment:
Yes, otherwise the type can't be deduced unless you configure it explicitly with custom code. That is the reason in the case that if you use interface or inheritance to store things to mongo Spring stores automagically the class name of the instance that was used.